src/lib/artsy_jwt.ts

```diff
@@ -0,0 +1,10 @@
+import jwt from "jwt-simple"
+import config from "config"
+const { HMAC_SECRET } = config
+
+export const decodeArtsyJWT = token => {
+  if (!token) {
+    return null
+  }
+  return jwt.decode(token as string, HMAC_SECRET, true, "HS256")
```

Comment:
`HMAC_SECRET` would be different for different apps, for example if a token coming into this method was a token generated for calling Vortex it would have been signed with Vortex's `ClientApplication`'s `internal_secret` and if the token was what was used to access CMS it would have been signed with `Volt`'s `internal_secret`. My guess is this `jwt.decode(...)` would fail if you pass a Vortex token. Looks like current `HMAC` secret config variable is basically to generate Causality's JWT.

Reply:
@ashkan18 ahh ok, so this could break in production as it is. Could we work around this by keeping volt's secret in the config and decode with that? Or is there another way to only validate against user JWTs I think that could work too. `jwt.decode` seems to work with user JWTs (generated from gravity), so maybe we can only enforce this check for those.

I guess another broader question is if we can solve this without involving JWTs at all, I'd be open to any of these options.

Reply:
The user's ID (or roles) can be _decoded_ from the JWT without _validating_ its signature. It's only the validation that requires the secret. I would look at the implementation of this `decode` method to understand how, or the `jwt-simple` library probably has a helper to allow this.

Reply:
You're actually passing `true` as the 3rd argument (`noVerify`) already, so I think no secret should be required here.